modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -242,11 +242,10 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
     } yield peer
 
   private def validateSeedlist(peer: PeerToJoin): F[Unit] =
-    seedlist match {
-      case None => Applicative[F].unit
-      case Some(entries) =>
-        if (entries.contains(peer.id)) Applicative[F].unit else PeerNotInSeedlist(peer.id).raiseError[F, Unit]
-    }
+    if (seedlist.forall(_.contains(peer.id)))
+      Applicative[F].unit
+    else
+      PeerNotInSeedlist(peer.id).raiseError[F, Unit]
```

Comment:
if you actually touch this part, you can use `whenA`/`unlessA` method from Applicative.

![image](https://github.com/Constellation-Labs/tessellation/assets/4959931/b8cbeb15-1aa5-4d3f-a7ad-7157972d8f78)